packages/syft/src/syft/service/sync/diff_state.py

```diff
@@ -1255,38 +1308,41 @@ class SyncInstruction(SyftObject):
     mockify: bool
 
     @classmethod
-    def from_widget_state(
+    def from_batch_decision(
         cls,
+        diff: ObjectDiff,
         sync_direction: SyncDirection,
-        widget: Any,
+        share_private_data: bool,
+        mockify: bool,
         decision: SyncDecision,
         share_to_user: SyftVerifyKey | None,
     ) -> Self:
         # read widget state
-        diff = widget.diff
         new_permissions_low_side = []
 
         # read permissions
         if sync_direction == SyncDirection.HIGH_TO_LOW:
             # To create read permissions for the object
             # job/usercode/request/TwinAPIEndpoint
-            if widget.share_private_data:  # or diff.object_type == "Job":
+            if share_private_data:  # or diff.object_type == "Job":
                 if share_to_user is None:
                     # job ran by another user
                     if not diff.object_type == "Job":
-                        raise ValueError("share_to_user is required for private data")
+                        raise ValueError(
+                            "share_to_user is required to share private data"
+                        )
                 else:
                     new_permissions_low_side = [
                         ActionObjectPermission(
-                            uid=widget.diff.object_id,
+                            uid=diff.object_id,
                             permission=ActionPermission.READ,
-                            credentials=share_to_user,  # type: ignore
+                            credentials=share_to_user,
                         )
                     ]
 
-        mockify = widget.mockify
-        if widget.has_unused_share_button:
-            print("Share button was not used, so we will mockify the object")
+        # TODO move this to the widget
```

Comment:
what about this?

Reply:
I dont think we need this print anymore, this only happens in case of a TwinAPIObject (and we added warning there)